test/hotspot/jtreg/compiler/lib/template_framework/StructuralName.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.template_framework;
+
+import java.util.List;
+
+/**
+ * {@link StructuralName}s represent things like method and class names, and can be added to the local
+ * scope with {@link Template#addStructuralName}, and accessed with {@link Template#structuralNames}, from where
+ * count, list or even sample random {@link StructuralName}s. Every {@link StructuralName} has a {@link StructuralName.Type},
+ * so that sampling can be restricted to these types.
+ *
+ * For field and variable names and alike, there are the analogous {@link DataName}s.
+ *
+ * @param name The {@link String} name used in code.
+ * @param type The type of the {@link StructuralName}.
+ * @param weight The weight of the {@link StructuralName}, it corresponds to the probability of choosing this
+ *               {@link StructuralName} when sampling later on.
+ */
+public record StructuralName(String name, StructuralName.Type type, int weight) implements Name {
+
+    /**
+     * Creates a new {@link StructuralName}.
+     */
+    public StructuralName {
+    }
+
+    /**
+     * The interface for the type of a {@link StructuralName}.
+     */
+    public interface Type extends Name.Type {
+        /**
+         * The name of the type, that can be used in code.
+         *
+         * @return The {@link String} representation of the type, that can be used in code.
+         */
+        String name();
+
+        /**
+         * Defines the subtype relationship with other types, which is used to filter {@link StructuralName}s
+         * in {@link View#exactOf}, {@link View#subtypeOf}, and {@link View#supertypeOf}.
+         *
+         * @param other The other type, where we check if it is the supertype of {@code 'this'}.
+         * @return If {@code 'this'} is a subtype of {@code 'other'}.
+         */
+        boolean isSubtypeOf(StructuralName.Type other);
+    }
```

Comment:
This is identical to `DataName.Type`. What is the benefit of having separate interfaces `DataName.Type` and `StructuralName.Type`? Couldn't we just move `isSubtypeOf()` directly to the `Name.Type` interface and use that one below and for the fields and expose that one instead to the user? This would mean that you can update all `DataName/StructuralName.Type` to `Name.Type`. I have not checked if this is fully possible but it just occurred to me when reviewing this duplicated interface now.